### PR TITLE
Make JIT work on iPhone armv7

### PR DESCRIPTION
**Pull request by dnfield**
Currently, debug mode apps fail to run on iPhone 4 (and presumably iPhone 5).  AOT is fine.

Without this flag, the Dart VM issues assembly instructions that aren't allowed on armv7.

A test for this would ideally be setting up one of these devices in our devicelab. We don't have any infra available for that at the moment.  /cc @digiter

### CODE REVIEW

File: shell/platform/darwin/ios/framework/Source/FlutterDartProject.mm [hunk 1]
```diff
@@ -140,6 +144,18 @@
...
+
+  // Tell Dart in JIT mode to not use integer division on armv7
+  // Ideally, this would be detected at runtime by Dart.
+  // TODO(dnfield): Remove this code  https://github.com/dart-lang/sdk/issues/24743
+  size_t size;
+  cpu_type_t type;
+
+  size = sizeof(type);
+  sysctlbyname("hw.cputype", &type, &size, NULL, 0);
+  if (type == CPU_TYPE_ARM) {
```
Comment (aam) [inline shell/platform/darwin/ios/framework/Source/FlutterDartProject.mm]: Does this have to be a runtime check?
Wouldn't it be possible to handle this similarly to how `kDartWriteProtectCodeArgs` is set up: https://github.com/flutter/engine/blob/master/runtime/dart_vm.cc#L319

Comment (dnfield) [reply shell/platform/darwin/ios/framework/Source/FlutterDartProject.mm]: We could do this as a compile define, but I couldn't quite find the one for ARM vs ARM64. I don't think the one you linked is it (but the overall method there could work).

**Comment by digiter** [comment]
I'm curious how would certain code do integer division if that's disabled? We can only do floating point division and round up?

**Comment by dnfield** [comment]
It's about which ASM instructions are used - integer division is still possible, but it will be handled using different (presumably less efficient if we were on an architecture that supported them) instructions

File: shell/platform/darwin/ios/framework/Source/FlutterDartProject.mm (hunk 1 above)
Comment (dnfield) [reply shell/platform/darwin/ios/framework/Source/FlutterDartProject.mm]: I can add the define - not sure if I'm doing it in the ideal way but will push that up.

Comment (dnfield) [reply shell/platform/darwin/ios/framework/Source/FlutterDartProject.mm]: Changed this to a feature define

**Comment by sjindel-google** [comment]
Thanks for fixing this!

File: common/config.gni [hunk 2]
```diff
@@ -58,3 +58,7 @@ if (is_fuchsia && using_fuchsia_sdk) {
 if ((is_ios || is_mac) && defined(enable_bitcode)) {
   flutter_enable_bitcode = enable_bitcode
 }
+
+if (target_cpu == "arm") {
+  feature_defines_list += [ "TARGET_CPU_ARMV7=1" ]
```
Comment (aam) [inline common/config.gni]: Do you need to define your own TARGET_CPU_ARMV7? I thought you could reuse existing TARGET_CPU_ARM

Comment (dnfield) [reply common/config.gni]: I didn't see TARGET_CPU_ARM defined anywhere.

Comment (dnfield) [reply common/config.gni]: Well either way it's workign - using that define instead.

Comment (dnfield) [reply common/config.gni]: I take it back, it's not working - that define is there for ARM64 as well

Comment (dnfield) [reply common/config.gni]: I was using `#ifdef` instead of `#if`.  Thanks for the help on this here and offline.

**Pull request merged**